library/std/src/thread/available_concurrency.rs

```diff
@@ -0,0 +1,155 @@
+use crate::io;
+use crate::num::NonZeroUsize;
+
+/// Returns the number of hardware threads available to the program.
+///
+/// This value should be considered only a hint.
+///
+/// # Platform-specific behavior
+///
+/// If interpreted as the number of actual hardware threads, it can undercount on
+/// Windows systems with more than 64 hardware threads. If interpreted as the
+/// available concurrency for that process, it can overcount on Windows systems
+/// when limited by a process wide affinity mask or job object limitations.
```

Comment:
On linux it may also overcount due to cgroup limits or affinity.

Reply:
Good point; updated the text to include that.